lib/plugins/aws/invokeLocal/index.js

```diff
@@ -148,9 +164,166 @@ class AwsInvokeLocal {
 
     const configuredEnvVars = this.getConfiguredEnvVars();
 
-    _.merge(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    const promises = _.entries(configuredEnvVars).map(([name, value]) => {
+      return this.resolveResource(value).then(output => {
+        configuredEnvVars[name] = output;
+      });
+    });
+
+    return BbPromise.all(promises).then(() => {
+      Object.assign(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    });
+  }
+
+  loadExportsImpl(token, cfExports) {
+    if (!cfExports) {
+      cfExports = [];
+    }
+    return this.provider
+      .request('CloudFormation', 'listExports', { NextToken: token })
+      .then(result => {
+        cfExports.push(...result.Exports);
+        if (result.NextToken) {
+          return this.loadExports(result.NextToken, cfExports);
+        }
+
+        return cfExports;
+      });
+  }
+
+  resolveResourceImpl(value) {
+    if (!_.isObject(value)) {
+      return new Promise(resolve => resolve(value));
+    }
+    return BbPromise.try(() => {
+      return new Promise((resolve, reject) => {
+        let resourceName = null;
+        let action = null;
+
+        if (value.Ref) {
+          action = 'Ref';
+          resourceName = value[action];
+        } else if (value['Fn::GetAtt']) {
+          action = 'Fn::GetAtt';
+
+          // Handle condition where GetAtt has Resource.Property format
+          if (typeof value[action] === 'string') {
+            value[action] = value[action].split('.');
+          }
+          if (value[action].length !== 2) {
+            reject(new ServerlessError(`Attribute definition unexpected ${JSON.stringify(value)}`));
+          }
+
+          // Get the resource name
+          resourceName = value[action][0];
+        } else if (value['Fn::ImportValue']) {
+          action = 'Fn::ImportValue';
+          resourceName = value[action];
+        } else {
+          return resolve(value);
+        }
+
+        if (action === 'Fn::ImportValue') {
+          resolve(
+            this.loadExports().then(cfExports => {
+              const result = cfExports.find(exp => exp.Name === resourceName);
+              if (!result) {
+                throw new ServerlessError(`Could not find import value for ${resourceName}`);
+              }
+              return result.Value;
+            })
+          );
+        } else {
+          let resource = this.serverless.service.resources.Resources[resourceName];
```

Comment:
I don't think such resolution can be seen as reliable, I indicated that in previous review: https://github.com/serverless/serverless/pull/7147#discussion_r366204868 (comment was marked as resolved, but remained not answered in any way)

Reply:
I modified the resolution if the resource was not found to work with the ability to specify a name of a function, rather than just inferring the name of the function.  Since the name element is not automatically generated in the serverless model, this gives the best immediate resolution possible.  Lines 241 - 245 does the custom name resolution, while lines 248 - 262 does the inferred name resolution.  I'm open to another way of doing this, but with the current serverless model how it is, this appears to be the most immediate way with the least impact

Reply:
> this appears to be the most immediate way with the least impact

In my eyes this approach is incomplete and problematic maintanance-wise. Resources may be further modified/added by the plugins, and resolution rules, options used, may change in a framework.

Ideally this resolution should support whatever happens in a plugins and framework, without a need of revisiting the logic when something changes. In my understanding the only way to achieve it, is to resolve resources directly from a generated template. Then also resolution is a one liner, which maintenance way cannot be better.

The problem here is, that packaging was never made when processing local invocation, and without introducing this step, we cannot inspect CF template. Still, that means that we should conditionally introduce packing when resolving env vars for local invocation

Reply:
I rewrote this to rely solely on the getCFResources results.  This will do two things.  First it will ensure the resources are actually created and deployed.  Second, it resolves the issue of the plugins.

Reply:
I've rewritten the name resolution to solely use getCFResources rather than the cloudformation template.  This ensures the resources are actually deployed, and eliminates the need to generate a cloudformation template during debugging